### PR TITLE
Thermostat bac 002 alzb

### PR DESCRIPTION
**Pull request by juan11perez**
Requesting addition of [this](https://www.aliexpress.com/item/1005001441524917.html?spm=a2g0s.9042311.0.0.34534c4d0ZdyPu) thermostat

### CODE REVIEW

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -2868,6 +2868,75 @@ const converters = {
...
+                        {saturday_p3: value[18] + 'h:' + value[19] + 'm ' + value[20]/2 + '°C'},
+                        {saturday_p4: value[21] + 'h:' + value[22] + 'm ' + value[23]/2 + '°C'},
+                        {sunday_p1: value[24] + 'h:' + value[25] + 'm ' + value[26]/2 + '°C'},
+                        {sunday_p2: value[27] + 'h:' + value[28] + 'm ' + value[29]/2 + '°C'},
+                        {sunday_p3: value[30] + 'h:' + value[31] + 'm ' + value[32]/2 + '°C'},
+                        {sunday_p4: value[33] + 'h:' + value[34] + 'm ' + value[35]/2 + '°C'},
+                    ],
+                };
+            case tuya.dataPoints.state: // Thermostat on standby = OFF, running = ON
+                return {system_mode: value ? 'cool' : 'off'};
```
Comment (Koenkk) [inline converters/fromZigbee.js:2898]: I've compared this converter with `fz.moes_thermostat`, the only difference I could find was this line.

Replacing `return {system_mode: value ? 'heat' : 'off'};` with below will make this converter obsolete.

```js
if (model.model === 'BAC-002-ALZB') {
  return {system_mode: value ? 'heat' : 'off'};
} else {
  return {system_mode: value ? 'cool' : 'off'};
}
```

File: devices.js [hunk 2]
```diff
@@ -15018,7 +15018,23 @@ const devices = [
             .withSystemMode(['off', 'heat', 'auto'], ea.STATE_SET).withLocalTemperatureCalibration(ea.STATE_SET)
             .withRunningState(['idle', 'heat'], ea.STATE).withAwayMode()],
     },
-
+    // HGKG
+    {
+        fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_dzuqwsyg'}],
+        model: 'BAC-002-ALZB',
```
Comment (Koenkk) [inline devices.js:15024]: When searching for `BAC-002-ALZB` on Google I cannot find anything about this device, where can it be bought?

**Comment by Koenkk** [comment]
Note to self, after this merge https://github.com/Koenkk/zigbee2mqtt.io/pull/681

File: devices.js (hunk 2 above)
Comment (juan11perez) [reply devices.js:15024]: Thank you. I've modified as per your instruction (however I think you meant as below) and removed the converter.
Also added the associated change to its reference in devices.js.

`if (model.model === 'BAC-002-ALZB') {
  return {system_mode: value ? 'cool' : 'off'};
} else {
  return {system_mode: value ? 'heat' : 'off'};
}`

Purchased from Aliexpress. 
https://www.aliexpress.com/item/1005001441524917.html?spm=a2g0s.9042311.0.0.34534c4d0ZdyPu
https://www.aliexpress.com/item/1005001482329022.html?spm=a2g0s.9042311.0.0.27424c4dshq3Ag

File: converters/fromZigbee.js (hunk 1 above)
Comment (juan11perez) [reply converters/fromZigbee.js]: Thank you.
I've modified fz.moes_thermostat per your instruction and removed fz.hgkh_thermostat

**Comment by juan11perez** [comment]
@Koenkk thank you for your time and guidance

**Pull request merged**